src/mpi/comm/comm_impl.c

```diff
@@ -711,42 +819,17 @@ int MPIR_Comm_get_name_impl(MPIR_Comm * comm_ptr, char *comm_name, int *resultle
 int MPIR_Comm_group_impl(MPIR_Comm * comm_ptr, MPIR_Group ** group_ptr)
 {
     int mpi_errno = MPI_SUCCESS;
-    int i, lpid, n;
-    int comm_world_size = MPIR_Process.comm_world->local_size;
     MPIR_FUNC_TERSE_STATE_DECL(MPID_STATE_MPIR_COMM_GROUP_IMPL);
 
     MPIR_FUNC_TERSE_ENTER(MPID_STATE_MPIR_COMM_GROUP_IMPL);
-    /* Create a group if necessary and populate it with the
-     * local process ids */
+    /* Create a local group if necessary */
     if (!comm_ptr->local_group) {
-        n = comm_ptr->local_size;
-        mpi_errno = MPIR_Group_create(n, group_ptr);
+        mpi_errno = comm_create_local_group(comm_ptr);
         MPIR_ERR_CHECK(mpi_errno);
```

Comment:
Refactoring of `comm_create_local_group ` should be in a separate commit

Reply:
done